src/astral/data-utils.js

```diff
@@ -0,0 +1,86 @@
+const tokenCache = {
+    access: null,
+    apiKey: null,
+    refresh: null,
+    expires: null
+};
+
+const _getTokenFromDb = () => new Promise((resolve, reject) => {
+    const req = indexedDB.open("firebaseLocalStorageDb", 1);
+    console.log("Created db req", req)
+    req.onsuccess = (event) => {
+        const db = req.result;
+        const trans = db.transaction("firebaseLocalStorage", IDBTransaction.READ)
+        const objStore = trans.objectStore("firebaseLocalStorage");
+
+        const cursorReq = objStore.openCursor();
+
+        cursorReq.onsuccess = (event) => {
+            console.log("Create cursor req", cursorReq.result)
+            resolve(cursorReq.result.value.value.stsTokenManager);
+        };
+    }
+
+    req.onerror = reject;
+});
+
+const getAccessToken = async () => {
+    if (!tokenCache.expires) {
+        const tokenData = await _getTokenFromDb();
+
+        tokenCache.access = tokenData.accessToken;
+        tokenCache.apiKey = tokenData.apiKey;
+        tokenCache.expires = tokenData.expirationTime;
+        tokenCache.refresh = tokenData.refreshToken;
+    } else if (tokenCache.expires < Date.now()) {
+        const tokenData = await (await fetch('https://securetoken.googleapis.com/v1/token?key=' + tokenCache.apiKey, {
+            method: 'POST',
+            headers: {'content-type': 'application/x-www-form-urlencoded'},
+            body: 'grant_type=refresh_token&refresh_token=' + tokenCache.refresh,
+        })).json();
+        
+        tokenCache.access = tokenData.access_token;
+        tokenCache.refresh = tokenData.refresh_token;
+        const tokenInfo = jwt_decode(tokenCache.access);
```

Comment:
If this is all you need, you shouldn't have to depend on jwt_decode.
You can get the info like this `JSON.parse(atob(tokenCache.access.split(".")[1]))` in theory.. which is better/shorter than importing that an external dependency just to call `atob` and `JSON.parse`

Reply:
I'll take a look, to be fair I have seen the logic in jwt_decode to have a couple of fail safes and assumed they were needed, if in our case, that's all we need, I'll change it.